src/components/Kv/KvPhoneInput.vue

```diff
@@ -0,0 +1,201 @@
+<template>
+	<div class="kv-phone-input"
+		:class="{
+			'kv-phone-input--is-empty': isEmpty,
+			'kv-phone-input--is-invalid': !isValid,
+			'kv-phone-input--is-valid': isValid,
+		}"
+	>
+		<div class="kv-phone-input__wrapper">
+			<label
+				class="show-for-sr"
+				:for="`country_select_${id}`"
+			>
+				Select your country
+			</label>
+
+			<div class="kv-phone-input__select-wrapper">
+				<kv-flag
+					class="kv-phone-input__select-flag"
+					:country="selectedCountryCode"
+					:inline-svg="true"
+				/>
+
+				<select
+					class="kv-phone-input__select-country"
+					:id="`country_select_${id}`"
+					v-model="selectedCountryCode"
+					@change="onInputCountry"
+				>
+					<option
+						v-for="country in countryList"
+						:key="`country-${country.name}`"
+						:value="country.code"
+					>
+						{{ country.name }}
+					</option>
+				</select>
+			</div>
+
+			<span class="kv-phone-input__country-prefix">
+				+{{ countryCallingCode }}
+			</span>
+
+			<input type="tel"
+				class="kv-phone-input__input"
+				:id="id"
+				:placeholder="placeholderNumber"
+				:value="displayNumber"
+				@input="onInputPhoneNumber"
+			>
+		</div>
+	</div>
+</template>
+
+<script>
+import { getCountryList } from 'flag-icon-css';
+import {
+	AsYouType,
+	getCountryCallingCode,
+	getCountries as getSupportedCountryCodes,
+	getExampleNumber,
+	parsePhoneNumberFromString,
+} from 'libphonenumber-js';
+import exampleNumbers from 'libphonenumber-js/examples.mobile.json'; // used for populating placeholders
+import KvFlag from '@/components/Kv/KvFlag';
+
+const supportedCountryCodes = getSupportedCountryCodes();
+const countryList = getCountryList() // get all country names and codes
+	.filter(countryObj => supportedCountryCodes.includes(countryObj.code)) // remove unsupported countries
+	.sort((a, b) => a.name.localeCompare(b.name)); // alphabetize
+
+export default {
+	components: {
+		KvFlag,
+	},
+	model: {
+		prop: 'value',
+		event: 'input'
+	},
+	props: {
+		id: {
+			type: String,
+			required: true
+		},
+		value: {
+			type: String,
+			default: ''
+		},
+	},
+	data() {
+		return {
+			countryList,
+			displayNumber: this.formatPhoneNumber(this.value), // pretty display of the phone number
+			selectedCountryCode: 'US',
+		};
+	},
+	computed: {
+		countryCallingCode() {
+			return getCountryCallingCode(this.selectedCountryCode);
+		},
+		e164Number() { // E.164 formatted phone number
+			const phoneNumber = parsePhoneNumberFromString(this.displayNumber, this.selectedCountryCode);
+			return phoneNumber?.number || '';
+		},
+		isEmpty() {
+			return this.displayNumber.length === 0;
+		},
+		isValid() {
+			const phoneNumber = parsePhoneNumberFromString(this.displayNumber, this.selectedCountryCode);
```

Comment:
And move this code to the parent validator

Reply:
Well probably not as that method comes out of the lib imported here...

Seems like this component should house the basic validation based on the lib you're using. Then if we need to do additional validation based on our own data or some other criteria that would be either plugged-in here or dealt with in a/the wrapping component.

Reply:
@mcstover good call 👍